.github/workflows/stale.yml

```diff
@@ -0,0 +1,26 @@
+name: Check stale issues and pull requests
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 12 * * *'
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v5
+        with:
+          debug-only: true
+          repo-token: ${{ secrets.PROJECT_BOARD_AUTOMATION }}
+          stale-issue-message: 'This issue is stale because it has been open for 30 days with no activity. It will be closed in 7 days unless the stale label is removed.'
+          stale-issue-label: stale
+          stale-pr-message: 'This pull request is stale because it has been open for 30 days with no activity. It will be closed in 7 days unless the stale label is removed.'
+          stale-pr-label: stale
+          days-before-stale: 30
```

Comment:
When is it counted from on first run? Will the bot just mass-message almost all of existing issues when we run it for the first time? :)

Reply:
Good question. I could not see mention of it in their docs. But yes, I suppose that is when you commit the config to your repo, so the workflow gets registered. As the PR is currently a draft, it will not appear there.

Honestly, I do not know how they keep the counter for each issue/PR and how efficient it is to run it for a big project. But it would be nice to find out, indeed. I know that they have an option to limit the number of operations: [operation-per-run](https://github.com/actions/stale#operations-per-run), so it can indeed dispatch a bunch of actions, and we need to limit it to our needs to not exceed the the github rate limit.

I do not know precisely how it will perform in a repository with many issues. However, we can test it safely while the debug is true. It will not label or close any issue or PR, just display in the workflow logs what actions it would take.

Here is the output of a test that I run:
```
Starting the stale action process...
Processing the batch of issues  #1  containing  5  issues...
[#20] Pull request #20
[#9] Issue #9
[#8] Issue #8
[#4] Issue #4
[#2] Issue #2
Batch  #1  processed.
No more issues found to process. Exiting...
Statistics:
Processed items: 5
├── Processed issues: 4
└── Processed PRs   : 1
New stale PRs: 1
Added PRs labels: 1
Added PRs comments: 1
Fetched items: 5
Fetched items events: 5
Fetched items comments: 5
Fetched pull requests: 1
Operations performed: 15
```

Reply:
The downside is that we'll need to wait with enabling it until we're done with labeling our issue backlog. Currently most of our issues do not have any desirability label and would be marked stale and then closed. Maybe we'd need some label to keep them open after all... or maybe we should have some minimum date?

Reply:
> 
> That might be good but won't it just look at when an issue was last modified instead?
> 
> Well, I guess mass tagging as stale would not be bad as long as we are sure the rules are set up correctly and it would only touch issues we do consider stale. We definitely need that dry run. Too bad that log does not have issue titles. This will make it quite tedious to review.

Sorry, I guess I express myself wrong. The counter is per issue/PR; yes, it checks when they were last modified. I meant that the workflow would start the counters when we deploy it (changing it from draft to normal PR will already trigger the execution - but it is with debugging enabled, so no worries). Then it will display the statistics according to our configuration.
 
However, I could not understand one thing in their source code: where they store the counters. It seems they only keep the last states in memory and iterate over all issues/PR updating or removing the counter accordingly. But I guess this level of reliability is fine. In the worst-case scenario, you would only wait for more to close something.

> Too bad that log does not have issue titles. This will make it quite tedious to review.

About having the issue titles, we could fork the original project and adapt it. But you will know the issues for sure when they get tagged. They are only not shown in the statistics. And we can have another automation to move tagged `stale` issues to some column in the board (like the icebox). However, the apps that I found available that can do this do not have a dry-run mode.

> The downside is that we'll need to wait with enabling it until we're done with labeling our issue backlog. Currently most of our issues do not have any desirability label and would be marked stale and then closed. Maybe we'd need some label to keep them open after all... or maybe we should have some minimum date?

I will enable it in the dry run, it will not harm, and we can already see some statistics. We can first use it only for some labeled tags (e.g., bug) and see how it plays out. For that, we can use the configuration `only-labels`.

Reply:
> It seems they only keep the last states in memory

Is it a github app? I thought it runs within a github action and in that case memory is not persistent.

> About having the issue titles, we could fork the original project and adapt it.

We shouldn't do that. We don't want the burden of having to maintain the fork. If anything submitting a PR would be much better.

> But you will know the issues for sure when they get tagged.

Well, the point was to see that before we allow them to get tagged.

Anyway, we can probably work around that somehow. For example I see that in the `#<number>` format so maybe just pasting the output into github comment preview would give us the titles...